### PR TITLE
[master] Add heap dump docs for CC Router and Use placeholder for Choreo Connect version

### PR DESCRIPTION
**Pull request by renuka-fernando**
## Purpose
Port https://github.com/wso2/docs-apim/pull/6067
Port https://github.com/wso2/docs-apim/pull/6107
Port https://github.com/wso2/docs-apim/pull/6112

### CODE REVIEW

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-apis-as-immutable-gateway.md [hunk 1]
```diff
@@ -5,6 +5,7 @@ For that, the apictl projects needs to be mounted to the `docker-compose/resourc
 (default location) of the adapter. 
 At the startup of the adapter, it picks the projects and deploys them. After startup, APIs should be added using apictl.
 
+For deploying APIs as immutable gateway using Helm chart, please refer [Deploy APIs as Immutable Gateway in Production Deployment]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline/#deploy-apis-as-immutable-gateway).
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-apis-as-immutable-gateway.md:8]:
```suggestion
For deploying APIs as an immutable Gateway using Helm chart, please refer [Deploy APIs as Immutable Gateway in Production Deployment]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline/#deploy-apis-as-immutable-gateway).
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md [hunk 2]
```diff
@@ -112,7 +112,7 @@ Follow the steps given below to configure how your Choreo Connect deployment sho
 1.  Open the `values.yaml` file in the `<KUBERNETES_HOME>/helm/choreo-connect` directory of your local copy.
 
     !!! Info
-        Before you do any changes, go through the [default configurations](https://github.com/wso2/kubernetes-microgateway/tree/v1.1.0.1/helm/choreo-connect) in this file.
+        Before you do any changes, go through the [default configurations](https://github.com/wso2/kubernetes-microgateway/tree/{{choreo_connect.helm_chart.git_tag}}/helm/choreo-connect) in this file.
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md:115]: @renuka-fernando  do they have to manually specify the tag in the link?

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/troubleshooting/debug-memory.md [hunk 3]
```diff
@@ -37,3 +37,113 @@ Follow the steps below to take a heap dump of the Adapter.
 ## Taking a heap dump of Enforcer
 
 Enforcer is implemented in Java and it is configured with required parameters to dump the heap information when out of memory errors happen. The dump can be collected from `/home/wso2/logs/heap-dump.hprof`
+
+## Taking a heap dump of Router
+
+Router uses the Envoy Proxy as the core component that does the traffic routing. The default Router which is used in you production environment is not supporting heap profiling and we have built a separate Router Docker image which enables heap profiling. Follow the steps below to take a heap profile of the Router.
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/troubleshooting/debug-memory.md:43]:
```suggestion
The Router uses the Envoy Proxy as the core component that does the traffic routing. The default Router that is used in your production environment does not support heap profiling and we have built a separate Router Docker image that enables heap profiling. Follow the steps below to take a heap profile of the Router.
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md (hunk 2 above)
Comment (renuka-fernando) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md:115]: We have specified it in the mkdocs.yaml file, I have specified it

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/troubleshooting/debug-memory.md [hunk 4]
```diff
@@ -37,3 +37,113 @@ Follow the steps below to take a heap dump of the Adapter.
...
+The Router uses the Envoy Proxy as the core component that does the traffic routing. The default Router that is used in your production environment does not support heap profiling and we have built a separate Router Docker image that enables heap profiling. Follow the steps below to take a heap profile of the Router.
+
+### Prerequisites
+
+1.  Install [pprof](https://github.com/google/pprof#readme) and [Graphviz](http://www.graphviz.org/) in your local machine. **pprof** is used to analyze the heap profile and **Graphviz** enables to analyze them with graphs.
+
+2.  Build ***Router Debug Image***, if you are taking heap profile **on top of an updated Router image**.
+
+    !!! note
+        For the **GA version** of the Router docker image you can use `wso2am/choreo-connect-router:{{choreo_connect.version}}-debug-{{choreo_connect.envoy_version}}` as the ***Router Debug Image***.
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/troubleshooting/debug-memory.md:52]:
```suggestion
        For the **GA version** of the Router Docker image, you can use `wso2am/choreo-connect-router:{{choreo_connect.version}}-debug-{{choreo_connect.envoy_version}}` as the **Router Debug Image**.
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/troubleshooting/debug-memory.md [hunk 5]
```diff
@@ -37,3 +37,113 @@ Follow the steps below to take a heap dump of the Adapter.
...
+### Prerequisites
+
+1.  Install [pprof](https://github.com/google/pprof#readme) and [Graphviz](http://www.graphviz.org/) in your local machine. **pprof** is used to analyze the heap profile and **Graphviz** enables to analyze them with graphs.
+
+2.  Build ***Router Debug Image***, if you are taking heap profile **on top of an updated Router image**.
+
+    !!! note
+        For the **GA version** of the Router Docker image, you can use `wso2am/choreo-connect-router:{{choreo_connect.version}}-debug-{{choreo_connect.envoy_version}}` as the **Router Debug Image**.
+
+    1.  Create a Dockerfile as follows. Here we create a custom docker image by replacing the envoy binary in default docker image with heap profile enabled envoy binary.
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/troubleshooting/debug-memory.md:54]:
```suggestion
    1.  Create a Dockerfile as follows. Here we create a custom Docker image by replacing the envoy binary in default Docker image with heap profile enabled envoy binary.
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/troubleshooting/debug-memory.md [hunk 6]
```diff
@@ -37,3 +37,113 @@ Follow the steps below to take a heap dump of the Adapter.
...
+        FROM <ROUTER_DOCKER_IMAGE_WITH_WSO2_UPDATES>
+        COPY --from=wso2am/choreo-connect-router:{{choreo_connect.version}}-debug-{{choreo_connect.envoy_version}} /usr/local/bin/envoy /usr/local/bin/envoy
+        ```
+
+        ```Dockerfile tab="Sample Dockerfile"
+        FROM wso2/choreo-connect-router:{{choreo_connect.version}}.1 # for update level 1
+        COPY --from=wso2am/choreo-connect-router:{{choreo_connect.version}}-debug-{{choreo_connect.envoy_version}} /usr/local/bin/envoy /usr/local/bin/envoy
+        ```
+
+    2. Build an docker image and push it to a Docker registry that your K8s cluster can pull it.
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/troubleshooting/debug-memory.md:68]:
```suggestion
    2. Build a Docker image and push it to a Docker registry that your Kubernetes cluster can pull it.
```

**Review by samgnaniah — APPROVED** [review]

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md (hunk 2 above)
Comment (samgnaniah) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md:115]: Ok great. Thanks!

**Pull request merged**